trio/_core/tests/test_asyncgen.py

```diff
@@ -3,10 +3,15 @@
 import pytest
 from math import inf
 from functools import partial
-from async_generator import aclosing
+
 from ... import _core
 from .tutil import gc_collect_harder, buggy_pypy_asyncgens, restore_unraisablehook
 
+try:
+    from asyncgenerator import aclosing
+except ImportError:
+    aclosing = None
+
```

Comment:
My opinion is that this is not worth it (dealing with more conditionals in the tests, coverage, etc.).

It's reasonable to just skip tests needing aclosing for older python versions (that will be dropped in the coming years anyway).  pytest makes this easy.